persist/fs/msgpack/encoder.go

```diff
@@ -139,16 +157,35 @@ func (enc *Encoder) EncodeLogMetadata(entry schema.LogMetadata) error {
 	return enc.err
 }
 
-func (enc *Encoder) encodeIndexInfo(info schema.IndexInfo) {
+func (enc *Encoder) encodeIndexInfo(version int, info schema.IndexInfo) {
+	enc.encodeRootObject(version, indexInfoType)
+	if version == 1 {
+		enc.encodeIndexInfoV1(info)
+	} else if version == 2 {
+		enc.encodeIndexInfoV2(info)
+	} else {
+		panic(fmt.Sprintf("invalid index info version: %d", version))
```

Comment:
Can you just make this `enc.err = fmt.Errorf(...)` instead of panic?  We'd like to continue to really avoid panic at if at all possible.

Reply:
yep good idea